Client/Frontend/UserContent/UserScripts/RequestBlocking.js

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+window.braveBlockRequests = (args) => {
+  "use strict";
+  const securityToken = args.securityToken
+  const sendMessage = (resourceURL) => {
+    return webkit.messageHandlers[args.handlerName].postMessage({
+      securityToken: securityToken,
+      resourceURL: resourceURL.href,
+      sourceURL: window.location.href,
+      resourceType: 'xmlhttprequest'
+    })
+  }
+  
+  const { fetch: originalFetch } = window
+  window.fetch = function() {
+    const [resource, config] = arguments
+    
+    // Extract the url string
+    let urlString
+    if (resource instanceof String) {
+      // Simple send where resource is a string
+      urlString = resource
+    } else if (resource !== undefined && resource.url instanceof String) {
+      // Assume its a Request
+      urlString = resource.url
+    } else {
+      return originalFetch.apply(this, arguments)
+    }
+
+    const url = new URL(urlString, window.location.href)
+    return sendMessage(url).then(blocked => {
+      if (blocked) {
+        return new Promise(function(_, reject) {
+          reject(Error(`Content blocked for '${url}'`))
+        });
+      } else {
+        return originalFetch.apply(this, arguments)
+      }
+    })
+  }
+  
+  const originalOpen = XMLHttpRequest.prototype.open
+  XMLHttpRequest.prototype.open = function() {
+    const sendArguments = arguments
+    originalOpen.apply(this, arguments)
```

Comment:
Note that the return value here is needed. https://developer.mozilla.org/en-US/docs/Web/API/Window/open#return_value